### ISSUE TITLE
Restore consensus-reactor.md

### ISSUE THREAD

**Issue by greg-szabo** [cometbft/cometbft#2718]
## Bug Report
The `spec/reactors` folder was deleted back in 2021 in [this](https://github.com/tendermint/tendermint/commit/72d15a4b0773adb8d5160a34d9a8e2d981394484#diff-5531f9389977e6bcb98535af7a345e3d7d81d74fd53e726357a1a7b441bc12fb) commit. I'm not sure yet why; my hypothesis is that the information from those folders was disseminated into other documentation.

However, the content of `spec/reactors/consensus/consensus-reactor.md` doesn't seem to be available anywhere. I've checked the other documentation folders, the CometBFT documentation website, and the Tendermint documentation website.

It seems that reactor documentation and tutorials have been sparse lately.

Should we restore the whole `spec/reactors` folder or only the `consunsus-reactor.md` file?

Maybe there is a good reason why the folder was deleted and we don't need them anymore.

**Comment by greg-szabo** [comment]
The commit is part of the 0.35 branch. I wonder if they thought that the 0.35 implementation was different enough that the documentation was outdated.

**Comment by andynog** [comment]
I think you can find them [here](https://github.com/tendermint/tendermint/tree/1b2b24055c5e44c41347cd524a713f0eddb7881c/spec/reactors) but not sure if it was the latest version but seems to be the closest to when they were removed

**Comment by greg-szabo** [comment]
Thanks, I can find different versions of the files, some in the spec repo, some in tendermint. The problem I'm tackling is how and why they disappeared. The commit mentioned on top shows when they disappeared from the tendermint repo, but it doesn't tell us why.

If I want to restore the files, I can use the ones from the tendermint repo at the time they were deleted, (https://github.com/tendermint/tendermint/commit/72d15a4b0773adb8d5160a34d9a8e2d981394484)

or I can use the one you mentioned, (https://github.com/tendermint/tendermint/tree/1b2b24055c5e44c41347cd524a713f0eddb7881c/spec/reactors)
which is at least on the main branch, but four days older than the deletion commit

or one from the spec repo, (https://github.com/tendermint/spec/commit/32b811a1fb6e8b40bae270339e31a8bc5e8dea31)
which was supposedly copied over according to one of Thane's commits while preparing to move to 0.37 (https://github.com/tendermint/tendermint/commit/174783220e58cba4fbcd9095fa5f6e8bf9fad79b). The description of this commit says that it copies the files from the spec repo but in reality it didn't copy the files, only the README.md from the repo.

Interestingly, this last mention from Thane is the newest commit from 2022 but it refers to a very old spec repo commit from 2020.

So, which version should we import? The one that was mentioned latest in the tendermint repo, but is actually the oldest from the spec repo; the one that removed it from the tendermint repo but there's no proof it was ever merged to main or the one that's slightly older but there is proof it was ever on main.

**Comment by cason** [comment]
The `tendermint/spec` repository wasn't active when v0.37 was released. It was included in `tendermint/tendermint` before v0.36 was released. Then v0.35 and v0.36 were reverted, `master` was the original branch, `main` became the new one. Then `tendermint/tendermint` was (forced) rebased, pointing to v0.34.

**Comment by cason** [comment]
I am for restoring this content, but with some careful reviewing, since a lot of stuff could have changed. I can help with this, but I don't have much time for doing archeology, ehe.

**Comment by greg-szabo** [comment]
Thanks, Daniel, for the history. It looks like the last version is whatever is on `master` from the v0.36 days. However, `master` currently is missing the `reactors` folder. The folder was deleted when things moved to the `tendermint/spec` repo.

The only reliable commit that I found was when Marko and Tess _deleted_ the folder from `tendermint/tendermint` in https://github.com/tendermint/tendermint/commit/72d15a4b0773adb8d5160a34d9a8e2d981394484 . The files didn't all make it to the spec repository, and when things got reverted, not all the files were restored.

So, to keep all possible files in scope, I'd restore from the above link and go through the files. It is possible that we should add a "legacy documentation" warning or similar.

@cason can you confirm that the above plan would be the best? If you feel there's a newer commit we could use, I'm all for it. If you feel this is fine, I'll take it from here.

**Comment by sergio-mena** [comment]
@greg-szabo I agree with your approach, but would prefer to have @cason's green light here.

**Comment by cason** [comment]
Just a correction on your timeline. The content possibly was in `tendermint/tendermint` when it was removed, from the PR you pointed out. When I joined, it was in the `tendermint/spec`. In 2022, we archived this repository and moved everything (back?) to `tendermint/tendermint`. Then we had the reversion of `master` (v0.35 and initial v0.36), and certainly some content from the `spec/` folder was "lost" in the process. In particular, the PBTS spec that was on `main` until the beginning of this year was pretty outdated, and we "rescued" its latest version from `master`.

**Comment by cason** [comment]
I think that trying to follow or maintain the history of these files should not be really the point here. As you realize, the timeline is messy. A possible solution (again, we have done that in the PBTS branch), is to just restore the files into a new branch, then we review its content.

For instance, in https://github.com/cometbft/cometbft/pull/2013 I've manually migrated some content I found is some closed PR.

**Comment by cason** [comment]
Green light you have :)

But lets keep this as a draft PR for the moment, so we can at least review before adding back to the repository?

I like the idea of marking it as legacy, I've done the same with this content: https://github.com/cometbft/cometbft/tree/main/spec/p2p/legacy-docs. By the way, it was also written about at the same period as the specs you are pointing to.
